### PR TITLE
[Feature][Connector-V2][OceanBase] Support vector types on OceanBase

### LINKED ISSUE #7290: [Feature][Connector-V2][OceanBase] Support vector types on OceanBase
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

OceanBase will officially support vector types in the near future, and now there is a docker image for testing, we can support vector types in SeaTunnel based on it.  
```
docker pull oceanbase/oceanbase-ce:vector0.2
```

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by xxsc0529**
### Purpose of this pull request

Added support for vectors in Oceanbase Close #7290

### Does this PR introduce _any_ user-facing change?

yes

### How was this patch tested?

When creating test data in milvus, the database is default, the collection is created as simple_examole, and the data sample is as shown in the following figure
image
The test script is as follows
env {
  job.mode = "BATCH"
}

source {
  Milvus {
    url = "http://localhost:19530"
    token = "root:Milvus"
    database = "default"
    collection="simple_example"
  }
}

transform {
}

sink {
  jdbc {
    url = "jdbc:oceanbase://localhost:2881/test"
    driver = "com.oceanbase.jdbc.Driver"
    user = "root"
    password = ""
    generate_sink_sql =true
    compatible_mode="mysql"
    database = "test"
    table = "simple_example"
  }
}
Since oceanbase-client does not support vector parsing, oceanbase catalog does not support automatic creation of vector tables
The DDL of Oceanbase is as follows:
CREATE TABLE IF NOT EXISTS simple_example
                (
                  `book_id` int NOT NULL,
                  `book_intro` vector(4) DEFAULT NULL,
                  `book_title` varchar(64) DEFAULT NULL,
                   primary key (`book_id`)
               );

image
image
This mission was successful

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by Hisoka-X** [comment]
Could you add a test case for insert/read vector into/from oceanbase?

**Comment by xxsc0529** [comment]
> Could you add a test case for insert/read vector into/from oceanbase?

I added the test case and the test on giothub was successful
image
Because the milvus container and oceanbase container occupy too much memory after startup, @Disable annotations are added

**Comment by Hisoka-X** [comment]
Thanks @xxsc0529 ! How about waiting #7401 be merged to replace milvus in test case?

**Comment by xxsc0529** [comment]
> Thanks @xxsc0529 ! How about waiting #7401 be merged to replace milvus in test case?

ok

**Comment by xxsc0529** [comment]
> Thanks @xxsc0529 ! How about waiting #7401 be merged to replace milvus in test case?

@Hisoka-X this #7401 is merged

**Comment by Hisoka-X** [comment]
Please merge from dev then use fake source to produce vector data to write into oceanbase

**Comment by xxsc0529** [comment]
> Please merge from dev then use fake source to produce vector data to write into oceanbase
The test case of fakesource has been added, and the result is as follows
image

**Comment by xxsc0529** [comment]
> Please merge from dev then use fake source to produce vector data to write into oceanbase
What should I do if I find that the connection fails to execute the workflow?
The link is below
https://paimon.apache.org/docs/0.6/maintenance/configurations/#coreoptions
location
./docs/en/connector-v2/sink/Paimon.md
image
@Hisoka-X

**Comment by Hisoka-X** [comment]
> > Please merge from dev then use fake source to produce vector data to write into oceanbase
> > What should I do if I find that the connection fails to execute the workflow?
> > The link is below
> > https://paimon.apache.org/docs/0.6/maintenance/configurations/#coreoptions
> > location
> > ./docs/en/connector-v2/sink/Paimon.md
> > image
> > @Hisoka-X

try to fix it in #7435

**Comment by Hisoka-X** [comment]
Please merge from dev. Thanks

**Comment by xxsc0529** [comment]
> Please merge from dev. Thanks

done,
image
Did you change the logic of mysqlcdc? There is an error here  @Hisoka-X

**Comment by Hisoka-X** [comment]
> Did you change the logic of mysqlcdc? There is an error here @Hisoka-X

No, maybe it's unstable. Please retry failed CI.

**Comment by xxsc0529** [comment]
> > 是不是改了mysqlcdc的逻辑，这里报错@Hisoka-X
> 
> 不是，可能是不稳定，请重试失败的 CI。

ok

**Comment by xxsc0529** [comment]
> > Did you change the logic of mysqlcdc? There is an error here @Hisoka-X
> 
> No, maybe it's unstable. Please retry failed CI.

ok,done, @Hisoka-X

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcOceanBaseMilvusIT.java [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+import java.util.stream.Stream;
+
+import static org.awaitility.Awaitility.given;
+
+@Slf4j
+@DisabledOnContainer(
+        value = {},
+        type = {EngineType.SPARK, EngineType.FLINK},
+        disabledReason = "Currently SPARK and FLINK not support adapt")
+@Disabled("oceanbase vector and milvus takes up too much memory")
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcOceanBaseMilvusIT.java]: Please remove disabled on fake to oceanbase case.

Comment (xxsc0529) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcOceanBaseMilvusIT.java]: At present, Oceanbase's vector image will not be closed, resulting in a process still running, and this image problem will be dealt with in the future, but the function is okay。seatunne's executeJob logic has an assertion to determine whether there is an image running, and after testing, replacing it with a mysql image or other versions of the image will not have this error, and the current ob vector image has this problem, so add @Disable without modifying the original test logic on the premise。
image

Comment (Hisoka-X) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcOceanBaseMilvusIT.java]: > Oceanbase's vector image will not be closed, resulting in a process still running

Do you know the reason? And is there any problem if we don't solve this problem and enable the test case at the same time?

Comment (xxsc0529) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcOceanBaseMilvusIT.java]: > Please remove disabled on fake to oceanbase case.

Subsequent  PR will be submitted, the OB-related testcontainers will be modified, and the test container will be replaced to the latest version @Hisoka-X  What do you think

Comment (xxsc0529) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcOceanBaseMilvusIT.java]: > > Oceanbase's vector image will not be closed, resulting in a process still running
> 
> Do you know the reason? And is there any problem if we don't solve this problem and enable the test case at the same time?

The problem of the ob vector container, I am sure that there is no problem with the test case, and the synchronous results can be printed out at the moment, but the test container still has threads running and cannot be closed

Comment (Hisoka-X) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcOceanBaseMilvusIT.java]: ~~ok.~~

Comment (Hisoka-X) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcOceanBaseMilvusIT.java]: > At present, Oceanbase's vector image will not be closed, resulting in a process still running, and this image problem will be dealt with in the future, but the function is okay。seatunne's executeJob logic has an assertion to determine whether there is an image running, and after testing, replacing it with a mysql image or other versions of the image will not have this error, and the current ob vector image has this problem, so add @disable without modifying the original test logic on the premise。 image

This is the assert logic provided by seatunnel to ensure that there are no abnormal threads in the engine. I believe this test case did not pass our assert logic. The `JNA cleaner` thread come from oceanbase client. This is a bug come from jna tool. Fixed by https://github.com/java-native-access/jna/commit/e8182b28e8baaaeae7b68767bca91c1012e39d4d in version [5.14.0](https://github.com/java-native-access/jna/releases/tag/5.14.0)
So we should do this:
1. open test case and add `JNA cleaner` thread into https://github.com/apache/seatunnel/blob/570bbb39a0de9deed60e4089823b80bc59ec3263/seatunnel-e2e/seatunnel-e2e-common/src/test/java/org/apache/seatunnel/e2e/common/container/seatunnel/SeaTunnelContainer.java#L407
2. waiting oceanbase client update jna version to 5.14.0 then remove it from `isIssueWeAlreadyKnow`

Comment (Hisoka-X) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcOceanBaseMilvusIT.java]: cc @whhe as well.

**Review by Hisoka-X — APPROVED** [review]
LGTM. Thanks @xxsc0529 !

**Review by hailin0 — APPROVED** [review]

**Pull request merged**